Tests/StreamChatTests/WebSocketClient/WebSocketClient_Tests.swift

```diff
@@ -500,9 +501,18 @@ final class WebSocketClient_Tests: XCTestCase {
         XCTAssertFalse(eventsBatcher.mock_processImmediately.called)
         
         // Simulate disconnection
-        webSocketClient.disconnect()
+        let expectation = expectation(description: "disconnect completion")
+        webSocketClient.disconnect {
+            expectation.fulfill()
+        }
         
         // Assert `processImmediately` is triggered
         AssertAsync.willBeTrue(eventsBatcher.mock_processImmediately.called)
+        
+        // Simulate batch processing completion
+        eventsBatcher.mock_processImmediately.calls.last?()
+        
+        // Assert completion called
+        wait(for: [expectation], timeout: 0)
```

Comment:
We need to provide a timeout. Depending on the load in the CI, this might take more time.
```suggestion
        wait(for: [expectation], timeout: 0.5)
```

Reply:
The `completion` is synchronously called on the previous line `eventsBatcher.mock_processImmediately.calls.last?()` so the expectation has been already fulfilled at this point.

The same is valued for all other places where `timeout: 0` is used